src/reconcile/rec_visibility.c

```diff
@@ -763,12 +780,20 @@ __rec_fill_tw_from_upd_select(
          * visible. In this case, the on page value is not appended. Verify that.
          */
         if (last_upd->next != NULL) {
-            WT_ASSERT_ALWAYS(session,
-              last_upd->next->txnid == vpack->tw.start_txn &&
-                last_upd->next->start_ts == vpack->tw.start_ts &&
-                last_upd->next->type == WT_UPDATE_STANDARD && last_upd->next->next == NULL,
-              "Tombstone is globally visible, but the tombstoned update is on the update "
-              "chain");
+            if (F_ISSET(S2C(session), WT_CONN_RECOVERING))
+                WT_ASSERT_ALWAYS(session,
+                  last_upd->next->txnid == WT_TXN_NONE &&
+                    last_upd->next->start_ts == vpack->tw.start_ts &&
+                    last_upd->next->type == WT_UPDATE_STANDARD && last_upd->next->next == NULL,
+                  "Tombstone is globally visible, but the tombstoned update is on the update "
+                  "chain");
+            else
+                WT_ASSERT_ALWAYS(session,
+                  last_upd->next->txnid == vpack->tw.start_txn &&
+                    last_upd->next->start_ts == vpack->tw.start_ts &&
+                    last_upd->next->type == WT_UPDATE_STANDARD && last_upd->next->next == NULL,
+                  "Tombstone is globally visible, but the tombstoned update is on the update "
+                  "chain");
```

Comment:
Can we have one dedicated assert to the `txnid` as it is the only different component between the two asserts? Or use the ternary operator and have one assert?

Reply:
I think the assert about the txn id needs to be with the assert of the timestamps. This is to ensure we are asserting that is globally visible. Asserting just the txnid doesn't tell you a story.